`python3Packages.jax` build failure on x86_64-linux as of `cfcfea5d`

## Issue description
Build of `python3Packages.jax` failed on x86_64-linux as of cfcfea5dbe319af077fb01c872c52b121077ea82.

[build log](https://github.com/samuela/nixpkgs-upkeep/actions/runs/1897553360)

Maintainer cc: @samuela
Other cc: n/a

### Steps to reproduce
1. Checkout nixpkgs at commit cfcfea5dbe319af077fb01c872c52b121077ea82
2. Run `nix-build -A python3Packages.jax`

## Technical details
```
- system: `"x86_64-linux"`
 - host os: `Linux 5.11.0-1028-azure, Ubuntu, 20.04.3 LTS (Focal Fossa)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.6.1`
 - nixpkgs: `/nix/store/6v2wg6ycxzrqwn5g2pxfvx5700vddq9a-source`
```

## Misc.
This issue was automatically generated by [nixpkgs-upkeep](https://github.com/samuela/nixpkgs-upkeep).
- [CI workflow](https://github.com/samuela/nixpkgs-upkeep/actions/runs/1897553360) that created this issue.
- Internal tag: 1812514623349726664

Looks to be some test failures:
```
2022-02-25T13:01:30.4847926Z =================================== FAILURES ===================================
2022-02-25T13:01:30.4854909Z ___________ CustomLinearSolveTest.test_custom_linear_solve_cholesky ____________
2022-02-25T13:01:30.4908674Z [gw0] linux -- Python 3.9.10 /nix/store/afi0ysqw20yiiw2gr2d28dx40bc4ddf8-python3-3.9.10/bin/python3.9
2022-02-25T13:01:30.4939194Z 
2022-02-25T13:01:30.4965635Z self = <custom_linear_solve_test.CustomLinearSolveTest testMethod=test_custom_linear_solve_cholesky>
2022-02-25T13:01:30.5066167Z 
2022-02-25T13:01:30.5438494Z     def test_custom_linear_solve_cholesky(self):
2022-02-25T13:01:30.5454251Z     
2022-02-25T13:01:30.5466838Z       def positive_definite_solve(a, b):
2022-02-25T13:01:30.5483564Z         factors = jsp.linalg.cho_factor(a)
2022-02-25T13:01:30.5496232Z         def solve(matvec, x):
2022-02-25T13:01:30.5501494Z           return jsp.linalg.cho_solve(factors, x)
2022-02-25T13:01:30.5513704Z         matvec = partial(high_precision_dot, a)
2022-02-25T13:01:30.5518668Z         return lax.custom_linear_solve(matvec, b, solve, symmetric=True)
2022-02-25T13:01:30.5524566Z     
2022-02-25T13:01:30.5536476Z       rng = self.rng()
2022-02-25T13:01:30.5542617Z       a = rng.randn(2, 2)
2022-02-25T13:01:30.5553776Z       b = rng.randn(2)
2022-02-25T13:01:30.5558932Z     
2022-02-25T13:01:30.5564749Z       tol = {np.float32: 1E-3 if jtu.device_under_test() == "tpu" else 1E-5,
2022-02-25T13:01:30.5567844Z              np.float64: 1E-12}
2022-02-25T13:01:30.5572109Z       expected = jnp.linalg.solve(np.asarray(posify(a)), b)
2022-02-25T13:01:30.5574865Z       actual = positive_definite_solve(posify(a), b)
2022-02-25T13:01:30.5578847Z >     self.assertAllClose(expected, actual, rtol=tol, atol=tol)
2022-02-25T13:01:30.5581439Z 
2022-02-25T13:01:30.5581640Z tests/custom_linear_solve_test.py:246: 
2022-02-25T13:01:30.5584466Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-02-25T13:01:30.5588690Z jax/_src/test_util.py:999: in assertAllClose
2022-02-25T13:01:30.5589210Z     self.assertArraysAllClose(x, y, check_dtypes=False, atol=atol, rtol=rtol,
2022-02-25T13:01:30.5593249Z jax/_src/test_util.py:964: in assertArraysAllClose
2022-02-25T13:01:30.5598146Z     _assert_numpy_allclose(x, y, atol=atol, rtol=rtol, err_msg=err_msg)
2022-02-25T13:01:30.5599105Z _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
2022-02-25T13:01:30.5602287Z 
2022-02-25T13:01:30.5602822Z a = array([-1422.5638, -3412.8064], dtype=float32)
2022-02-25T13:01:30.5606436Z b = array([-1422.8795, -3413.5635], dtype=float32), atol = 1e-05, rtol = 1e-05
2022-02-25T13:01:30.5610757Z err_msg = ''
2022-02-25T13:01:30.5610945Z 
2022-02-25T13:01:33.9285690Z     def _assert_numpy_allclose(a, b, atol=None, rtol=None, err_msg=''):
2022-02-25T13:01:33.9297290Z       if a.dtype == b.dtype == _dtypes.float0:
2022-02-25T13:01:33.9307772Z         np.testing.assert_array_equal(a, b, err_msg=err_msg)
2022-02-25T13:01:33.9310296Z         return
2022-02-25T13:01:33.9313804Z       a = a.astype(np.float32) if a.dtype == _dtypes.bfloat16 else a
2022-02-25T13:01:33.9318242Z       b = b.astype(np.float32) if b.dtype == _dtypes.bfloat16 else b
2022-02-25T13:01:33.9318620Z       kw = {}
2022-02-25T13:01:33.9323487Z       if atol: kw["atol"] = atol
2022-02-25T13:01:33.9323802Z       if rtol: kw["rtol"] = rtol
2022-02-25T13:01:33.9327957Z       with np.errstate(invalid='ignore'):
2022-02-25T13:01:33.9328445Z         # TODO(phawkins): surprisingly, assert_allclose sometimes reports invalid
2022-02-25T13:01:33.9331984Z         # value errors. It should not do that.
2022-02-25T13:01:33.9332377Z >       np.testing.assert_allclose(a, b, **kw, err_msg=err_msg)
2022-02-25T13:01:33.9332758Z E       AssertionError: 
2022-02-25T13:01:33.9336231Z E       Not equal to tolerance rtol=1e-05, atol=1e-05
2022-02-25T13:01:33.9336982Z E       
2022-02-25T13:01:33.9341222Z E       Mismatched elements: 2 / 2 (100%)
2022-02-25T13:01:33.9341550Z E       Max absolute difference: 0.7570801
2022-02-25T13:01:33.9345355Z E       Max relative difference: 0.00022186
2022-02-25T13:01:33.9345955Z E        x: array([-1422.5638, -3412.8064], dtype=float32)
2022-02-25T13:01:33.9350610Z E        y: array([-1422.8795, -3413.5635], dtype=float32)
2022-02-25T13:01:33.9350921Z 
2022-02-25T13:01:33.9351106Z jax/_src/test_util.py:185: AssertionError
2022-02-25T13:01:33.9351463Z =========================== short test summary info ============================
2022-02-25T13:01:33.9364874Z FAILED tests/custom_linear_solve_test.py::CustomLinearSolveTest::test_custom_linear_solve_cholesky
2022-02-25T13:01:33.9389436Z ========== 1 failed, 13955 passed, 1409 skipped in 1829.55s (0:30:29) ==========
2022-02-25T13:01:37.5374285Z error: builder for '/nix/store/7z1ydr7xc19108r3nlp5na4m45izascl-python3.9-jax-0.3.1.drv' failed with exit code 1;
2022-02-25T13:01:37.5382799Z        last 10 log lines:
2022-02-25T13:01:37.5386855Z        > E       Mismatched elements: 2 / 2 (100%)
2022-02-25T13:01:37.5390172Z        > E       Max absolute difference: 0.7570801
2022-02-25T13:01:37.5395569Z        > E       Max relative difference: 0.00022186
2022-02-25T13:01:37.5396988Z        > E        x: array([-1422.5638, -3412.8064], dtype=float32)
2022-02-25T13:01:37.5401687Z        > E        y: array([-1422.8795, -3413.5635], dtype=float32)
2022-02-25T13:01:37.5402168Z        >
2022-02-25T13:01:37.5406269Z        > jax/_src/test_util.py:185: AssertionError
2022-02-25T13:01:37.5408311Z        > =========================== short test summary info ============================
2022-02-25T13:01:37.5414536Z        > FAILED tests/custom_linear_solve_test.py::CustomLinearSolveTest::test_custom_linear_solve_cholesky
2022-02-25T13:01:37.5415113Z        > ========== 1 failed, 13955 passed, 1409 skipped in 1829.55s (0:30:29) ==========
```
I suspect this is due to the [overlay](https://github.com/samuela/nixpkgs-upkeep/blob/a2b5db7aa617277134620ed701df81f8321cf661/overlay.nix) since jax hasn't been changed otherwise.

--- a/pkgs/development/python-modules/jax/default.nix
+++ b/pkgs/development/python-modules/jax/default.nix
@@ -1,8 +1,10 @@
 { lib
 , absl-py
+, blas
 , buildPythonPackage
 , fetchFromGitHub
 , jaxlib
+, lapack
 , numpy
 , opt-einsum
 , pytestCheckHook
@@ -12,6 +14,9 @@
 , typing-extensions
 }:
 
+let
+  usingMKL = blas.implementation == "mkl" || lapack.implementation == "mkl";
+in
 buildPythonPackage rec {
   pname = "jax";
   version = "0.3.1";
@@ -58,6 +63,9 @@ buildPythonPackage rec {
     "-W ignore::DeprecationWarning"
     "tests/"
   ];
+
+  # See https://github.com/google/jax/issues/9705.
+  disabledTests = lib.optionals usingMKL [ "test_custom_root_with_aux" ];
 
   pythonImportsCheck = [
     "jax"

